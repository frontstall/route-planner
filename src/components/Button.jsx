import React from 'react';
import cn from 'classnames';

class Button extends React.Component {
  onClick = () => {
    const { onClick } = this.props;

    if (onClick) {
      onClick();
    }
  };

  render() {
    const {
      text = '',
      type = 'button',
      className = '',
      disabled = false,
    } = this.props;

    const classes = cn('button', className);

    return (
      <button
        className={classes}
        type={type}
        onClick={this.onClick}
        disabled={disabled}
      >
        {text}
      </button>
    );
  }
}

export default Button;
