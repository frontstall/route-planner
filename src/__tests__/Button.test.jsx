import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../components/Button';

it('renders button', () => {
  const component = renderer.create(<Button text="button" />).toJSON();
  expect(component).toMatchSnapshot();
});
