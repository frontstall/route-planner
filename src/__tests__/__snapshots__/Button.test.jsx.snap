// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  button
</button>
`;
